database/migrations/2024_03_23_180709_site2user_fk_and_index.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+use Illuminate\Database\Migrations\Migration;
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Support\Facades\DB;
+use Illuminate\Support\Facades\Schema;
+
+return new class extends Migration {
+    /**
+     * Run the migrations.
+     */
+    public function up(): void
+    {
+        if (Schema::hasTable('site2user')) {
+            echo "Adding siteid foreign key to site2user table...";
+            $num_deleted = DB::delete('DELETE FROM site2user WHERE siteid NOT IN (SELECT id FROM site)');
+            echo $num_deleted . ' invalid rows deleted' . PHP_EOL;
+            Schema::table('site2user', function (Blueprint $table) {
+                $table->foreign('siteid')->references('id')->on('site')->cascadeOnDelete();
+                $table->unique(['userid', 'siteid']);
+                $table->unique(['siteid', 'userid']);
```

Comment:
what's the motivation for adding a unique constraint for both permutations of these fields?

Reply:
`->unique()` adds a unique index, so the idea was to index it such that lookups in both directions are efficient.